kv/kvserver/rangefeed: TestBudgetReleaseOnProcessorStop failed

kv/kvserver/rangefeed.TestBudgetReleaseOnProcessorStop [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11568242?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11568242?buildTab=artifacts#/) on master @ [0f70a7007724cc6550324ad9dd635c31cec8e400](https://github.com/cockroachdb/cockroach/commits/0f70a7007724cc6550324ad9dd635c31cec8e400):


```
=== RUN   TestBudgetReleaseOnProcessorStop
    processor_test.go:1143: stream failed with error before all data was consumed retry rangefeed (REASON_SLOW_CONSUMER)
--- FAIL: TestBudgetReleaseOnProcessorStop (0.00s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBudgetReleaseOnProcessorStop.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31103

--- a/pkg/kv/kvserver/rangefeed/processor_test.go
+++ b/pkg/kv/kvserver/rangefeed/processor_test.go
@@ -1105,7 +1105,7 @@ func TestBudgetReleaseOnProcessorStop(t *testing.T) {
 		PushTxnsInterval: pushTxnInterval,
 		PushTxnsAge:      pushTxnAge,
 		EventChanCap:     channelCapacity,
-		EventChanTimeout: time.Millisecond,
+		EventChanTimeout: 100 * time.Millisecond, // enough time for registration to process events
 		MemBudget:        fb,
 		Metrics:          NewMetrics(),
 	})

